src/ui/public/filter_bar/filter_bar.less

```diff
@@ -1,7 +1,7 @@
 @import (reference) "~ui/styles/variables";
 
 filter-bar {
-  z-index: 20 !important;
+  z-index: 2 !important;
```

Comment:
I have a suggestion which is a little bit more work, but it's very straightforward and will help us avoid this problem from popping up again in the future:

1. Let's set this to `z-index: @globalDepthFilterBar !important;`.
2. On line 158 of `styles/base.less` let's change `> kbn-top-nav { z-index: 3; }` to `> kbn-top-nav { z-index: @globalDepthLocalNav; }`.
3. Let's create `styles/variables/depth.less`.
4. Let's import this new file into `styles/variables.less` like so: `@import "~ui/styles/variables/depth.less";`.
4. Inside this file, we'll assign both of these variables:

```less
/**
 * 1. The local nav contains tooltips which should pop over the filter bar.
 */
@globalDepthFilterBar: 2; /* 1 */
@globalDepthLocalNav: 3; /* 1 */
```

This way the relationship of these two depths will be really obvious and if someone reintroduces this bug by changing their values, the cause should be readily apparent.